text/0000-tracked-properties.md

```diff
@@ -227,34 +248,350 @@ memoization) if a property shows up as a bottleneck during profiling.
+
+```js
+const Config = Service.extend({
+  polling: {
+    shouldPoll: false,
+    pollInterval: -1,
+  },
+
+  init() {
+    this._super(...arguments);
+
+    fetch('config/api/url')
+      .then(r => r.json())
+      .then(polling => set(this, 'polling', polling));
+  }
+})
+
+class SomeComponent extends Component {
+  @service config;
+
+  @tracked
+  get pollInterval() {
+    // When we access shouldPoll and pollInterval, they don't have tags or
+    // wrapped getters. We have no idea that they could update in the future.
+    let shouldPoll = this.get('config.polling.shouldPoll');
+    let pollInterval = this.get('config.polling.pollInterval');
+
+    return shouldPoll ? pollInterval : -1;
+  }
+}
+```
+
+This solution will _only_ be necessary in cases where users do not know ahead of
+time whether a property is A. immutable, B. tracked, or C. a computed property.
+As time goes on and more and more services, addons, and apps are converted to
+`@tracked`, the need for `Ember.get` will become smaller and smaller, until it
+can be removed from the framework entirely (along with `Ember.set`).
+
+### Computed Property Interop with Tracked
+
+The previous section covers the cases where classic Ember properties are
+accessed from tracked properties, but we also have to define the behavior for
+when the opposite occurs: tracked properties are accessed by from a classic
+context.
+
+There are two major ways this can occur:
+
+1. Computed Properties
+2. Observers
+
+Observers are strictly within the old paradigm, so tracked properties will _not_
+attempt to interoperate with them. Using `Ember.set` on a tracked property will
+still notify that the property has updated, and any observers for that property
+will fire accordingly.
+
+Computed properties, on the other hand, are similar enough to tracked properties
+that interoperability would be useful. As such, computed properties will _also_
+automatically add tracked properties to their dependencies as they are accessed.
+This will use the same mechanism as tracked properties do with the autotrack
+stack, and will allow users to progressively convert from computed properties to
+tracked properties over time by whittling away at the explicit dependencies
+within a CP.
+
+### Interop with EmberObject
+
+While `@tracked` is designed from the ground up for native Javascript classes,
+it will have to interoperate with legacy code for some time. In the previous
+sections, we noted how `get` would have to be used with any legacy code which
+has not been updated to tracked properties. As such, to make the transition as
+easy as possible, `tracked` will also be made to work with the classic
+object-model:
+
+```js
+const Config = Service.extend({
+  polling: tracked({
+    value: {
+      shouldPoll: false,
+      pollInterval: -1,
+    }
+  }),
+
+  init() {
+    this._super(...arguments);
+
+    fetch('config/api/url')
+      .then(r => r.json())
+      .then(polling => set(this, 'polling', polling));
```

Comment:
One selling point about `tracked()` here is that you can actually drop the call to `set()` and just use `this.polling = polling`. I think we can sell `tracked()` here not as a sop to compatibility but an exciting improvement in its own right that allows us to finally get rid of `this.set()`, just like how we got rid of `this.get()`.

Reply:
Totally agree, we should emphasize this first and foremost. I think this example should still be included to show how backwards compatibility works when users _don't_ have control of the service.